structuredevent-service-legacy/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/StructuredEventToClusterShapeConverter.java

```diff
@@ -56,6 +56,7 @@
 
         if (blueprintDetails != null) {
             cdpClusterShape.setClusterTemplateName(defaultIfEmpty(blueprintDetails.getName(), ""));
+            cdpClusterShape.setClusterTemplateDetails(defaultIfEmpty(blueprintDetails.getBlueprintJson(), ""));
```

Comment:
Please avoid adding the whole blueprint json to the event, instead create a custom json string that contains only the required data, as explained in the jira ticket description.

The reason for this is that recently there were several occasions where the events become too long and were split into multiple records in elastic search, which caused the EDH collection pipeline to disregard the events. To hotfix this I added length restriction to longer fields (e.g. definition details) to keep the whole events around < 8000 chars. The original request in the jira ticket is reasonable, because it does not make a very long string (but it should be tested locally to be sure), but adding the whole blueprint would certainly make the event too long again.

Reply:
Adding only serviceTypes as json string to the clusterTemplateDetails.